### PR TITLE
Update LangVersion in individual projects

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: sdk/eventhub/Directory.Build.props [hunk 1]
```diff
@@ -24,6 +24,6 @@
     this needs to be set after the common properties import.
   -->
   <PropertyGroup Condition="$(MSBuildProjectName.StartsWith('Azure.')) and $(MSBuildProjectName.EndsWith('.Tests'))">
-    <LangVersion>9.0</LangVersion>
+    <LangVersion>11.0</LangVersion>
```
Comment (jsquire) [inline sdk/eventhub/Directory.Build.props:27]: If we don't need to explicitly pin anymore, then we can just remove this section (L18 - L28).    There was something that caused issues with C# 10 where we had to pin to 9.

**Review by jsquire — APPROVED** [review]

**Comment by pallavit** [comment]
Can/should we plan to add an analyzer rule to check for this?

File: sdk/eventhub/Directory.Build.props (hunk 1 above)
Comment (AlexanderSher) [reply sdk/eventhub/Directory.Build.props]: Done

**Comment by AlexanderSher** [comment]
> Can/should we plan to add an analyzer rule to check for this?

@pallavit You mean check that internal projects don't define lang version? From time to time it can be useful, cause new versions of language can introduce breaking changes that we may have no capacity to fix immediately.

**Comment by jsquire** [comment]
> Can/should we plan to add an analyzer rule to check for this?

I don't think so.  Individual packages should be able to deviate if needed.   Particularly this is useful when you need to back-pin a specific library to work around compiler issues.

File: common/SmokeTests/SmokeTest/SmokeTest.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFrameworks>net7.0;net6.0;net461</TargetFrameworks>
-    <LangVersion>latest</LangVersion>
+    <LangVersion>11.0</LangVersion>
```
Comment (weshaggard) [inline common/SmokeTests/SmokeTest/SmokeTest.csproj:5]: Can't this override be removed?

Comment (AlexanderSher) [reply common/SmokeTests/SmokeTest/SmokeTest.csproj:5]: It is in a sample project, so I want to leave it here in case someone just copy it as a whole

**Comment by heaths** [comment]
> > Can/should we plan to add an analyzer rule to check for this?
> 
> @pallavit You mean check that internal projects don't define lang version? From time to time it can be useful, cause new versions of language can introduce breaking changes that we may have no capacity to fix immediately.

As long as new syntactic language features can be built by the SDK in the root _global.json_. Language features that require a newer SDK should already fail the build since we target `netstandard2.0` at least. We do require public APIs to be the same across TFMs for now, and I have #33433 on my plate to hook up to help ensure that since the .NET team already has a solution for this (as well as backcompat checks - a replacement for what we're using now).

**Pull request merged**